src/Scheduler.php

```diff
@@ -201,6 +221,16 @@ public function schedule(Task $task, Frequency $frequency): self
 
             $now = new DateTime();
             $nextDue = $frequency->getNextDue($now);
+            if ($frequency->isExpired($nextDue)) {
+                // Give the running task a chance to finish its running operations before detaching it
+                Loop::futureTick(function () use ($task) {
```

Comment:
Create a new promise "consisting" of all open promises for the current task. Remove the task once the new promise is fulfilled or rejected. Otherwise, running instances of the task will be cancelled if they are not completed by the next tick.